runner/views/run_api_view.py

```diff
@@ -24,41 +26,91 @@ class RunApiViewSet(mixins.ListModelMixin,
     queryset = Run.objects.prefetch_related(Prefetch('port_set', queryset=
     Port.objects.select_related('run'))).order_by('-created_date').all()
 
-    serializer_class = RunSerializerFull
+    def get_serializer_class(self):
+        if self.action == 'list' or self.action == 'retrieve':
+            return RunApiListSerializer
+        else:
+            return RunSerializerFull
+
+    def fix_query_list(self,request_query,key_list):
+        query_dict = request_query.dict()
+        for single_param in key_list:
+            query_value = request_query.getlist(single_param)
+            if query_value:
+                if ',' in query_value[0]:
+                    query_value = query_dict.get(single_param)
+                    query_dict[single_param] = query_value.split(",")
+                else:
+                    query_dict[single_param] = query_value
+        return query_dict
 
+    def query_from_dict(self,query_filter,queryset,input_list):
+        for single_input in input_list:
+            key, val = single_input.split(':')
+            query = {query_filter % key: val}
+            queryset = queryset.filter(**query).all()
+        return queryset
+
+    @swagger_auto_schema(query_serializer=RunApiListSerializer)
     def list(self, request, *args, **kwargs):
-        queryset = time_filter(Run, request.query_params)
-        job_group = request.query_params.getlist('job_group')
-        if job_group:
-            queryset = queryset.filter(job_group__in=job_group).all()
-        status_param = request.query_params.get('status')
-        if status_param:
-            if status_param not in [s.name for s in RunStatus]:
-                return Response({'details': 'Invalid status value %s: expected values %s' % (status_param, [s.name for s in RunStatus])}, status=status.HTTP_400_BAD_REQUEST)
-            queryset = queryset.filter(status=RunStatus[status_param].value)
-        input_params = request.query_params.get('input')
-        if input_params:
-            inputs = input_params.split(',')
-            for input in inputs:
+        query_list_types = ['job_groups','request_ids','inputs','tags','jira_ids']
+        fixed_query_params = self.fix_query_list(request.query_params,query_list_types)
+        serializer = RunApiListSerializer(data=fixed_query_params)
+        if serializer.is_valid():
+            queryset = time_filter(Run, fixed_query_params)
+            job_groups = fixed_query_params.get('job_groups')
+            jira_ids = fixed_query_params.get('jira_ids')
+            status_param = fixed_query_params.get('status')
+            ports = fixed_query_params.get('ports')
+            tags = fixed_query_params.get('tags')
+            request_ids = fixed_query_params.get('request_ids')
+            created_before = fixed_query_params.get('created_before')
```

Comment:
What is the difference between this and `time_filter`? Also we don't have query params required in `time_filter` listed in RunApiListSerializer.

Reply:
Good catch, I missed that function. It should be fixed now.